src/components/fx/hover.js

```diff
@@ -689,7 +689,7 @@ function createHoverText(hoverData, opts, gd) {
             // strip out our pseudo-html elements from d.name (if it exists at all)
             name = svgTextUtils.plainText(d.name || '');
 
-            if(name.length > 15) name = name.substr(0, 12) + '...';
+            if(name.length > commonLabelOpts.namelength && commonLabelOpts.namelength > 0) name = name.substr(0, commonLabelOpts.namelength - 3) + '...';
```

Comment:
`-1` sounds good for "no truncation." I'm not sure *why* someone would want to use `namelength=0` to completely hide the name, rather than using `hoverinfo` but I bet someone will come up with a clever reason for it!